nerfstudio/model_components/renderers.py

```diff
@@ -104,13 +101,14 @@ def combine_rgb(
         if BACKGROUND_COLOR_OVERRIDE is not None:
             background_color = BACKGROUND_COLOR_OVERRIDE
         if background_color == "random":
-            # If background color is random, the predicted color is returned without blending,
-            # as if the background color was black.
-            return comp_rgb
+            if self.training:
```

Comment:
This breaks it, I believe. Currently, the blending is done later in `blend_background`. The reason is that we want to use the image with a random background for computing the loss fn, but then blend the image with the correct background for visualization. Everything without computing the accumulated colours twice. If this change is accepted, the computed colour will (for most models) be blended twice. 
Have you tried doing the change to the code and trying to train nerfacto with random bg on some data?